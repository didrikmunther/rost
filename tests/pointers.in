fn main() {
	let p = 100
	let a = 123
	let b: pointer<int> = &a
	let c: pointer<pointer<int>> = &b
	let d: pointer<int> = &p

	printf("%i\n", a) // a = 123
	printf("%i\n", *b) // a = 123
	printf("%i\n", **c) // a = 123
	printf("%i\n", *c == &a) // b == b
	printf("%i\n", d - b) // &p - &a == 8
	printf("%i\n", b - *c) // &a - b == b - b == 0
	printf("%i\n", *(&a + 8)) // p = 100
}