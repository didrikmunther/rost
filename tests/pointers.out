123
123
123
1
8
0
100